gke-app/gke.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import sys, os
+import time
+
+from kubernetes import client, config, utils
+import kubernetes.client
+from kubernetes.client.rest import ApiException
+
+
+class K8s(object):
+    """
+    This class handles creation and deletion of jobs in a kubernetes cluster.
+    """
+    def __init__(self):
+        # Set logging
+        logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+        # Setup K8 configs
+        # config.load_kube_config() # if running locally
+        config.load_incluster_config()
+        self.configuration = kubernetes.client.Configuration()
+        self.api_instance = client.BatchV1Api()
+
+    def generate_job_name(self):
+        time_split = str(time.time()).split('.')
+        name = "job-" + time_split[0] + "-" + time_split[1]
+        return name
+
+
+    def create_job_object(self, name, image, labels, backoff_limit, env):
+        # Configureate Pod template container
+        container = client.V1Container(
+            name=name,
+            image=image,
+            env=env
+        )
+        # Create and configurate a spec section
+        template = client.V1PodTemplateSpec(
+            metadata=client.V1ObjectMeta(labels=labels),
+            spec=client.V1PodSpec(restart_policy="Never", service_account_name="gke-app", containers=[container]))
```

Comment:
the K8s class `__init__()` should take a `service_account_name` as an initializer and set it as a class value. Then you can use `self.service_account_name` as a class value instead of hard-coding it in here.